libs/banglejs/jswrap_bangle.c

```diff
@@ -1275,19 +1283,23 @@ void peripheralPollHandler() {
     // wake on tap, for front (for Bangle.js 2)
 #ifdef BANGLEJS_Q3
     if ((bangleFlags&JSBF_WAKEON_TOUCH) && (tapInfo&2)) {
-      if (!(bangleFlags&JSBF_LCD_ON)) {
+      if (lcdPowerTimeout && !(bangleFlags&JSBF_LCD_ON)) {
         bangleTasks |= JSBT_LCD_ON;
         handled = true;
       }
-      if (!(bangleFlags&JSBF_LCD_BL_ON)) {
+      if (backlightTimeout && !(bangleFlags&JSBF_LCD_BL_ON)) {
         bangleTasks |= JSBT_LCD_BL_ON;
         handled = true;
       }
-      if (bangleFlags&JSBF_LOCKED) {
+      if (lockTimeout && bangleFlags&JSBF_LOCKED) {
         bangleTasks |= JSBT_UNLOCK;
         handled = true;
       }
     }
+    // ignore if locked
+    if (bangleFlags & JSBF_LOCKED) handled = false;
+    // extend timeouts when unlocked
+    inactivityTimer = 0;
```

Comment:
I'm a bit concerned that in this code, any 'tap' interrupt from the accelerometer will keep the bangle awake, regardless of settings? So if you're just driving along on a bumpy road and the Bangle gets bounced around a bit, it won't turn itself off...

Maybe this should be in `bangleFlags&JSBF_WAKEON_TOUCH` ?